src/spyglass/spikesorting/analysis/v1/group.py

```diff
@@ -10,73 +10,131 @@
+    definition = """
+    unit_filter_params_name: varchar(128)
+    ---
+    include_labels = Null: longblob
+    exclude_labels = Null: longblob
+    """
+
+    @property
+    def default_params(self):
+        return {
+            "unit_filter_params_name": "all_units",
+            "include_labels": [],
+            "exclude_labels": [],
+        }
+
+    @classmethod
+    def insert_default(cls, **kwargs):
+        """
+        Insert default parameter set for position determination
+        """
+        cls.insert1(
+            {**cls().default_params},
+            skip_duplicates=True,
+        )
+
+
 @schema
 class SortedSpikesGroup(SpyglassMixin, dj.Manual):
     definition = """
     -> Session
+    -> SortedSpikesGroupUnitSelectionParams
     sorted_spikes_group_name: varchar(80)
     """
 
-    class Units(SpyglassMixin, dj.Part):
+    class SortGroup(SpyglassMixin, dj.Part):
         definition = """
         -> master
-        -> UnitInclusionOutput.proj(unit_inclusion_merge_id='merge_id')
+        -> SpikeSortingOutput.proj(spikesorting_merge_id='merge_id')
         """
 
     def create_group(
         self,
         group_name: str,
         nwb_file_name: str,
-        unit_inclusion_merge_ids: list[str],
+        unit_filter_params_name: str = "all_units",
+        keys: list[dict] = [],
     ):
         group_key = {
             "sorted_spikes_group_name": group_name,
             "nwb_file_name": nwb_file_name,
+            "unit_filter_params_name": unit_filter_params_name,
         }
         self.insert1(
             group_key,
             skip_duplicates=True,
         )
-        self.Units.insert(
-            [
-                {
-                    "unit_inclusion_merge_id": id,
-                    **group_key,
-                }
-                for id in unit_inclusion_merge_ids
-            ],
-            skip_duplicates=True,
-        )
+        for key in keys:
+            self.SortGroup.insert1(
+                {**key, **group_key},
+                skip_duplicates=True,
+            )
 
     @staticmethod
-    def fetch_spike_data(
-        key: dict, time_slice: slice = None
-    ) -> list[np.ndarray]:
-        # TODO: provide labels for each unit
-        # TODO: filter out artifact times
-        unit_inclusion_merge_ids = (
-            SortedSpikesGroup.Units
-            & {
-                "nwb_file_name": key["nwb_file_name"],
-                "sorted_spikes_group_name": key["sorted_spikes_group_name"],
-            }
-        ).fetch("unit_inclusion_merge_id")
+    def filter_units(
+        labels: list[list[str]],
+        include_labels: list[str],
+        exclude_labels: list[str],
+    ) -> np.ndarray:
+        """
+        Filter units based on labels
+        """
+        include_labels = set(include_labels)
+        exclude_labels = set(exclude_labels)
+
+        include_mask = np.zeros(len(labels), dtype=bool)
+        for ind, unit_labels in enumerate(labels):
```

Comment:
It might be more efficient to use numpy here. I would consider `np.isin` and `np.unique`